### ISSUE TITLE
Add option to clip tab names in editor

### ISSUE THREAD

**Issue by OverLordGoldDragon** [spyder-ide/spyder#12980]
I don't really need the full file's name to know which one it is, exactly - an option to limit the number of characters would be helpful to have more tabs within a click's reach. **Examples**:

![image](https://user-images.githubusercontent.com/16495490/84092411-4c9c6680-aa08-11ea-95e1-c7b5d4bcc730.png)

 1. Clipping to (e.g.) `name_maxlen=7` (excluding `~.py`)
 2. If names cannot be uniquely resolved within `name_maxlen`, extend until a different char is met

The full names can be kept in the clickable dropdown or `Ctrl + Tab`.

**Comment by goanpeca** [comment]
Hi @OverLordGoldDragon 

Thanks for the proposal. I am personally not in favor of adding options for this little visual nitpicks. I prefer to do “the reasonable” thing by default.

Do you have examples of what otherIDEs do or other programs do?

Also have you checked what human interface  guidelines of different operating systems say on this topic?

Cheers!

**Comment by OverLordGoldDragon** [comment]
@goanpeca I don't know of any IDEs that do this, which is an opportunity for Spyder to pioneer. I wouldn't call it a 'nitpick' for the benefits it offers; it removes an overhead step in viewing, accessing, and organizing modules when many are open - and making programming faster and easier is an IDE's goal.

Unsure what's meant by "human interface guidelines", but my human-to-IDE interface is sure enhanced by this.

**Comment by goanpeca** [comment]
I meant this https://developer.apple.com/design/human-interface-guidelines/

Thanks again. Will leave open to gather some interest.

**Comment by ccordoba12** [comment]
@dalthviz, @jnsebgosselin, @impact27, @jitseniesen, what do you think about @OverLordGoldDragon's proposal?

**Comment by dalthviz** [comment]
Maybe it could be useful if you consistently follow a convention like _file_, _file_test_, _file__*_, although I think there are also cases where for example a user is just trying to explorer a code base he is not familiar with and seeing incomplete filenames could be troublesome.

**Comment by OverLordGoldDragon** [comment]
@dalthviz I agree on your named case, but as an "option" it can be turned off any time; I frequently work with long filenames I've opened dozens of times over, the few I haven't I could cover over to see the full name anyway. Also, my original idea was to have the name "fade" instead of inserting a placeholder character - but latter seems easier to implement and does the trick anyway.

Btw, PyCharm has an option to [hide file extensions](https://www.jetbrains.com/help/pycharm/settings-editor-tabs.html#), which'd also help; alternatively, "if not `.py`, then show" - but I've never had to work with non-`.py` in Spyder.
